File: Selenium/Project/EmergencyContact.java
package HRM;

import java.time.Duration;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.Reporter;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

public class EmergencyContact {
	
	static WebDriver driver;
	static WebDriverWait wait;
	
	@BeforeClass
	
	public void setUp() {
		
		driver = new ChromeDriver();
		driver.manage().window().maximize();
		driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login");
		driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
		wait = new WebDriverWait(driver,Duration.ofSeconds(2));
		
	}
	
	
	@Test(priority =1)
	@Parameters({"username", "password"})
	public void login_page(String username, String password) {
		
		Assert.assertEquals(driver.getTitle(), "OrangeHRM");
		driver.findElement(By.id("txtUsername")).sendKeys(username);
		driver.findElement(By.id("txtPassword")).sendKeys(password);
		driver.findElement(By.id("btnLogin")).click();
		Boolean logged_in = driver.findElement(By.xpath("//*[@id = 'welcome']")).isDisplayed();
		Assert.assertTrue(logged_in);	
	}
	
	@Test(priority =2)
	public void emergencyContactInfo() {
		
		wait.until(ExpectedConditions.elementToBeClickable(By.id("menu_pim_viewMyDetails")));
		driver.findElement(By.id("menu_pim_viewMyDetails")).click();
		driver.findElement(By.linkText("Emergency Contacts")).click();
		List<WebElement> mobileNumber = driver.findElements(By.xpath("//table/tbody/tr"));
		for(WebElement row : mobileNumber) {
			String number = row.findElement(By.xpath("./td[5]")).getText();
			System.out.println(number);
			Reporter.log(number);
		}
	}

	@AfterClass
	public void tearDown() {
		driver.quit();
	}
}

File: Selenium/Project/EmergencyContact.py
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait


def emergency_contact(driver, wait):
    wait.until(expected_conditions.element_to_be_clickable((By.ID, "menu_pim_viewMyDetails")))
    driver.find_element(By.ID, "menu_pim_viewMyDetails").click()
    driver.find_element(By.LINK_TEXT, "Emergency Contacts").click()
    mobile_numbers = driver.find_elements(By.XPATH, "//table/tbody/tr")
    for mobile_number in mobile_numbers:
        number = mobile_number.find_element(By.XPATH, "./td[5]").text
        print(number)

with webdriver.Chrome() as driver:
    driver.get("http://alchemy.hguy.co/orangehrm")
    title = driver.title
    print(title)

    driver.find_element(By.ID, "txtUsername").send_keys("orange")
    driver.find_element(By.ID, "txtPassword").send_keys("orangepassword123")
    driver.find_element(By.ID, "btnLogin").click()
    assert driver.find_element(By.XPATH, "//*[@id = 'welcome']").is_displayed();
    print(driver.find_element(By.TAG_NAME, "h1").text)
    wait = WebDriverWait(driver, 5)
    emergency_contact(driver,wait)

    driver.close()
